data/mods/Xedra_Evolved/monsters/changeling.json

```
@@ -214,6 +214,133 @@
+    "color": "green",
+    "aggression": 50,
+    "morale": 120,
+    "melee_skill": 7,
+    "melee_dice": 1,
+    "melee_dice_sides": 6,
+    "weakpoint_sets": [ "wps_humanoid_body", "wps_humanoid_body_armor", "wps_humanoid_open_helmet" ],
+    "dodge": 5,
+    "harvest": "fae_march_lord",
+    "families": [ "prof_wp_demihuman" ],
+    "vision_day": 50,
+    "vision_night": 10,
+    "path_settings": { "max_dist": 30, "allow_open_doors": true, "avoid_traps": true, "avoid_sharp": true },
+    "special_attacks": [
+      {
+        "type": "monster_attack",
+        "attack_type": "melee",
+        "id": "wild_huntsman_sword",
+        "cooldown": 1,
+        "move_cost": 80,
+        "damage_max_instance": [ { "damage_type": "cut", "amount": 25, "armor_penetration": 5 }, { "damage_type": "biological", "amount": 5 } ],
+        "effects": [ { "id": "effect_reduced_healing", "chance": 25, "duration": [ 3600, 10800 ] } ],
+        "condition": { "test_eoc": "is_disarmed" },
+        "hit_dmg_u": "%1$s slashes your %2$s with its sword!",
+        "hit_dmg_npc": "%1$s slashes <npcname>'s %2$s with its sword!",
+        "miss_msg_u": "%1$s tries to hit you, but you dodge!",
+        "miss_msg_npc": "%1$s tries to hit <npcname>, but they dodge!",
+        "no_dmg_msg_u": "%1$s slashes your %2$s without penetrating your armor.",
+        "no_dmg_msg_npc": "%1$s slashes <npcname>'s %2$s without penetrating their armor."
+      },
+      {
+        "id": "wild_huntsman_illumination_spell",
+        "type": "spell",
+        "spell_data": { "id": "wild_hunt_illumination_spell" },
+        "cooldown": 45,
+        "condition": { "not": { "npc_has_effect": "wild_hunt_illumination_spell" } },
+        "monster_message": "%1$s makes a gestures with one hand a glow springs up around you!"
+      }
+    ],
+    "death_drops": "changeling_wild_hunt_death_drops",
+    "anger_triggers": [ "FRIEND_DIED", "FRIEND_ATTACKED", "HURT" ],
+    "flags": [ "SEES", "HEARS", "WARM", "BASHES", "GROUP_BASH", "HUMAN", "FAE_CREATURE", "CAN_OPEN_DOORS", "PATH_AVOID_DANGER" ],
+    "armor": {
+      "bash": 22,
+      "cut": 36,
+      "stab": 16,
+      "cold": 5000,
+      "bullet": 36,
+      "electric": 2,
+      "pure": 160,
+      "xe_cold_iron_cut_damage": 0,
+      "xe_cold_iron_bash_damage": 0,
+      "xe_cold_iron_stab_damage": 0
+    }
+  },
+  {
+    "id": "mon_wild_hunt_hound",
+    "type": "MONSTER",
+    "name": { "str": "hound of the wild hunt" },
+    "description": "An enormous dog, a meter and a half high at the shoulder, with silvery fur and multicolored eyes.  It sniffs the air and looks around alertly.",
+    "default_faction": "changeling",
+    "bodytype": "dog",
+    "species": [ "CHANGELING" ],
+    "volume": "200000 ml",
+    "weight": "200000 g",
+    "hp": 175,
+    "speed": 140,
+    "material": [ "fae_flesh" ],
+    "symbol": "D",
+    "color": "light_gray",
+    "looks_like": "mon_direwolf",
+    "scents_tracked": [ "sc_human", "sc_fetid" ],
```

```suggestion
    "scents_tracked": [ "sc_human", "sc_fetid", "sc_fae", "sc_vampire" ],
```
Figured for long term planning those are needed.
Just realized there is no werewolf or therian scent?  Also only Undine have sc_fae? weird.  I think Arvore sometimes smell like flowers but probably should smell like fae sometimes?

---

Can a PC have multiple scents? I admit I didn't realize that XE added extra scents (Arvore don't smell like flowers until they get one of their traits--all the base traits ["Root and Branch" etc] should probably have the sc_fae scent)

---

I think generally when something happens you lose one scent and it's replaced by another.  I'll check that to confirm.

---

```
void Character::update_type_of_scent( bool init )
{
    scenttype_id new_scent = scent_sc_human;
    for( const trait_id &mut : get_functioning_mutations() ) {
        if( mut.obj().scent_typeid ) {
            new_scent = mut.obj().scent_typeid.value();
        }
    }

    if( !init && new_scent != get_type_of_scent() ) {
        get_scent().reset();
    }
    set_type_of_scent( new_scent );
}

void Character::update_type_of_scent( const trait_id &mut, bool gain )
{
    const std::optional<scenttype_id> &mut_scent = mut->scent_typeid;
    if( mut_scent ) {
        if( gain && mut_scent.value() != get_type_of_scent() ) {
            set_type_of_scent( mut_scent.value() );
            get_scent().reset();
        } else {
            update_type_of_scent();
        }
    }
}

void Character::set_type_of_scent( const scenttype_id &id )
{
    type_of_scent = id;
}

scenttype_id Character::get_type_of_scent() const
{
    return type_of_scent;
}
```

Yep looks like only one scent at a time

---

Werewolf at least, the scent can be put on the forms so they smell like a human as a human and a wolf as a wolf.